packages/utils/src/useSubscription.ts

```diff
@@ -127,28 +130,36 @@ export function useSubscription<TData, TError, R = TData>(
           if (observersCount === 0) {
             firestoreUnsubscribe(subscriptionHash);
           } else {
-            const isSubscribedToFirestore = !!firestoreUnsubscribes[subscriptionHash];
+            const isSubscribedToFirestore =
+              !!firestoreUnsubscribes[subscriptionHash];
             if (isSubscribedToFirestore) {
-              const cachedData = queryClient.getQueryData<TData | null>(queryKey);
+              const cachedData = queryClient.getQueryData<TData | null>(
+                queryKey
+              );
               const hasData = !!eventCount[subscriptionHash];
 
               if (hasData) {
                 resolvePromise(cachedData ?? null);
               }
             } else {
-              firestoreUnsubscribes[subscriptionHash] = subscribeFn(async (data) => {
-                eventCount[subscriptionHash] ??= 0;
-                eventCount[subscriptionHash]++;
-                if (eventCount[subscriptionHash] === 1) {
-                  resolvePromise(data || null);
-                } else {
-                  queryClient.setQueryData(queryKey, data);
+              firestoreUnsubscribes[subscriptionHash] = subscribeFn(
+                async (data) => {
+                  eventCount[subscriptionHash] ??= 0;
+                  eventCount[subscriptionHash]++;
+                  if (eventCount[subscriptionHash] === 1) {
+                    resolvePromise(data || null);
```

Comment:
```suggestion
                    resolvePromise(data ?? null);
```

Otherwise, if `data === false` you get back `null` and you never get the actual value of the data. 🤦🏼‍♂️

I spent hours trying to figure out why my DB value of `false` kept coming back `null`.